libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -360,6 +372,8 @@ void AP_ICEngine::update(void)
             // force ignition off when disarmed
             state = ICE_OFF;
         }
+        // reset the crank retry counter when disarmed
+        crank_retry_ct = 0;
```

Comment:
Should probably also reset the counter when we shut the ignition off. This way, someone can shut the engine off/on to reset if they want to force it to retry again.

Reply:
I am yet not sure of the approach of counter being reset is correct way so waiting for few more feedback. But rather then resetting at ignition off maybe we can reset it at Mavlink command of DO_ENGINE_CONTROL.
I will have a look at it.

Reply:
Agreed with a reset if the user moves away from the auto-restart position and back into it